File: Data Structures & Algorithms/duplicate-integer/submission-5.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        const hasSet = new Set();
        for (let i = 0; i < nums.length; i++) {
            if (!hasSet.has(nums[i])) {
                hasSet.add(nums[i]);
            } else {
                return true;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-5.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hasSet = {}

        for i, n in enumerate(nums):
            c = target - n
            if c not in hasSet:
                hasSet[n] = i
            else:
                return [hasSet[c], i]

            
